core/dictgen/src/rootcling_impl.cxx

```diff
@@ -4549,17 +4549,13 @@ int RootClingMain(int argc,
    std::ostream *dictStreamPtr = NULL;
    // Store the temp files
    tempFileNamesCatalog tmpCatalog;
-   if (!dictpathname.empty()) {
```

Comment:
> dictpathname is never empty because gOptDictionaryFileName is a required
argument.

That wasn't the case before, for some definition of before. @vgvassilev did you change that behavior, and was that change intentional?

I can get to `dictpathname.empty()` by passing `""` as argument, i.e. `rootcling "" Linkdef.h`. But that bombs before, I suppose? Would still be good to diagnose that.

Reply:
I am afraid that was a mistake on my part. Sorry. Thanks for catching it :)